torch/distributed/device_mesh.py

```diff
@@ -488,6 +325,162 @@ def __init__(
+                root_mesh, {}
+            )
+            # Let mesh_dim_name maps to the current layout whose backend has not been initialized.
+            # If mesh_dim_name has already mapped to another layout, we throw an error.
+            if name is not None:
+                if (
+                    name in names_to_layouts_map
+                    and names_to_layouts_map[name] != layout
+                ):
+                    raise ValueError(
+                        f"Mesh_dim_name {name} has already mapped to layout {layout}."
+                    )
+                names_to_layouts_map[name] = layout
+
+            # If the layout has already been mapped to a process group backend, we ignore the
+            # backend_override and group, and return early. We also throw a warning for that.
+            if layout in layouts_to_groups_map:
+                # TODO: We need to bookkeep pg_option together with the group name.
+                if backend_override != (None, None):
+                    warnings.warn(
+                        f"Group for {layout} ({name=}) already exists, ignoring backend override"
+                    )
+                if group is not None:
+                    warnings.warn(
+                        f"Group for {layout} ({name=}) already exists, ignoring explicit group"
+                    )
+                return
+
+            # When user explicitly pass in a process group, we directly reuse that PG as backend rather
+            # than creating a new one.
+            if group is not None:
+                layouts_to_groups_map[layout] = group.group_name
+                return
+
+            default_group = _get_default_group()
+
+            if layout == _Layout(((get_world_size(), 1),)) and backend_override == (
+                None,
+                None,
+            ):
+                # Append the default pg to the first dim groups only if the default pg is compatible with `self.device_type`.
+                # Otherwise, create new pg.
+                ranks = list(range(get_world_size()))
+                group = (
+                    new_group(
+                        backend="cpu:gloo,cuda:nccl",
+                        ranks=ranks,
+                        group_desc="mesh_default",
+                    )
+                    if torch.cuda.is_available()
+                    and get_backend(default_group) == "gloo"
+                    else default_group
+                )
+            else:
+                # Generate the pg_ranks_by_dim for pg_creation from the layout.
+                pg_ranks_by_dim = array_mesh if array_mesh is not None else layout.layout_to_global_ranks(get_world_size())
+                backend, pg_options = backend_override
+
+                # If we have a 2D mesh with mesh_dim_names ("dp", "tp"), the group description
+                # of the subgroups would be `mesh_dim_dp` and `mesh_name_tp`.
```

Comment:
typo, `mesh_name_tp`?  should be `mesh_tp` right? or is there a bug in the actual group_desc code below?